contents/computerVision/cv.md

```diff
@@ -0,0 +1,314 @@
+
+while(1):
+    ret, frame = cap.read()
+
+    if ret:
+      fgmask = fgbg.apply(frame)
+      cv2.imshow('frame',fgmask)
+      k = cv2.waitKey(30) & 0xff
+      if k == 27:
+          break
+    else:
+      break
+```
+
+Some results:
+
+Original frame:
+
+![Background not subtracted](object-tracking-example-1.jpg "Original Frame")
+
+With `BackgroundSubtractorMOG`:
+
+![Background subtracted](object-tracking-example-2.jpg "MOG")
+
+With `BackgroundSubtractorMOG2`:
+
+![Background subtracted](object-tracking-example-3.jpg "MOG2")
+
+With `BackgroundSubtractorGMG`:
+
+![Background subtracted](object-tracking-example-4.jpg "GMG")
+
+## GPU-accelerated CV
+
+For computer vision systems, digital images are often translated into matrices, whose nature depends on the chosen color presentation. For example, an image in the RGB (Red, Green, Blue) color space will be presented by 3 matrices, each contains the color intensity of the individual pixels with regards to red, green, or blue. There are other color representations, such as the CMYK or the HSV that annotates the pixels' Hue, Saturation, and Value.<sup>[[1]](#footnote1)</sup>
+
+Underneath, all computer vision algorithms deal with images as matrices, thus employ theories in Linear Algebra to process images. In computer hardware, there is a class of computing units that specializes in matrix operations called the Graphics Processing Units (GPUs) - their main purpose is to speed up heavy graphical tasks such as video rendering, gaming, or 3D modelling. Therefore, GPUs are well-suited to serve as computer vision computing units.<sup>[[11]](#footnote11)</sup>
+
+Most prominent computer vision libraries such as OpenCV, dlib, or VisionWorks support the use of GPUs in the computing process. For example, OpenCV has APIs that allow a kernel process running on the CPU to transfer image matrices to one or more GPUs to perform heavy functions such as Gaussian filtering or image stitching.<sup>[[12]](#footnote12)</sup> Hardware manufacturers have even tried to ease the image transferring process between CPU and GPU, by introducing system memory regions accessible to both CPU and GPU.
+
+## What's next
+
+There is an abundance of resources to learn and apply computer vision; however, not all of them are free or beginner-friendly.  The following are some great courses and tutorials freely accessible and aim for the masses:
+
+Before diving into computer vision, it is recommended that learner acquire basic familiarity in Python, C++, or Java, as most tutorials and courses in CV use one of those languages:
+
+[Python For Beginners](https://www.python.org/about/gettingstarted/)
+
+[Learn C++](http://www.learncpp.com/)
+
+[Java Tutorial](https://www.tutorialspoint.com/java/index.htm)
+
+It is also advisable that learner has some basic understandings in Linear Algebra.
+
+[Linear Algebra by Khan Academy](https://www.khanacademy.org/math/linear-algebra)
+
+**Computer vision:**
```

Comment:
It's a bit odd to see this sub heading in the middle of the paragraph. No harm removing it.

Reply:
In fact, you can tweak this whole section to present it as three steps:
>First, ... {programming}
>Next, you can ... {linear algebra}
>After that, ... {cv}